### PR TITLE
Signed links

### PR DESCRIPTION
**Pull request by elv-zenia**
Add CreateSignedLink method to append an authorization token to an existing link.

### CODE REVIEW

File: src/client/ContentManagement.js [hunk 1]
```diff
@@ -1380,6 +1380,73 @@ exports.CreateLinks = async function({
...
+   path: string (path to link target),
+   targetHash: string (optional, for cross-object links),
+   target: string (path to use in ctx/elv/lnk)
+ }
+
+ * @methodGroup Links
+ * @namedParams
+ * @param {string} libraryId - ID of the library
+ * @param {string} objectId - ID of the object
+ * @param {Array<Object>} links - Link specifications
```
Comment (elv-kevin) [inline src/client/ContentManagement.js:1401]: param definition for link doesn't match implementation

File: src/client/ContentManagement.js [hunk 2]
```diff
@@ -1380,6 +1380,73 @@ exports.CreateLinks = async function({
...
+ */
+exports.CreateSignedLink = async function({
```
Comment (elv-kevin) [inline src/client/ContentManagement.js:1403]: this method shouldn't be for updating existing links. it should create a link if not present, or overwrite the existing one appropriately (making sure extra unrelated fields in the link are preserved)

File: src/AuthorizationClient.js [hunk 3]
```diff
@@ -240,6 +242,47 @@ class AuthorizationClient {
...
+      methodName: "canEdit"
+    });
+
+    const { objectId } = Utils.DecodeVersionHash(versionHash);
+
+    if(!canEdit) {
+      throw Error(`Current user does not have permission to edit content object ${objectId}`);
+    }
+
+    const signerAddress = (this.client.signer && this.client.signer.address) || "";
```
Comment (elv-kevin) [inline src/AuthorizationClient.js:258]: use this.client.CurrentAccountAddress()

https://eluv-io.github.io/elv-client-js/ElvClient.html#CurrentAccountAddress

**Review by elv-kevin — CHANGES_REQUESTED** [review]

**Comment by elv-zenia** [comment]
@elv-kevin Updated

**Comment by elv-zenia** [comment]
@elv-kevin I updated `CreateLinks` based on current usage, which just cares about putting link metadata directly into a specified path. But since assets go in a path like `index`/`slug`/`link`, I made it so that the client is responsible for providing the right path.

**Pull request merged**